docker/dev/docker-compose.yml

```diff
@@ -1,75 +1,109 @@
-version: '3.3'
+version: "3.3"
+
+networks:
+  # communication to web and clients
+  lemmyexternalproxy:
+  # communication between lemmy services
+  lemmyinternal:
+    driver: bridge
+    internal: true
 
 services:
-  nginx:
+  lemmy-proxy:
     image: nginx:1-alpine
+    container_name: lemmy-proxy
+    networks:
+      - lemmyinternal
+      - lemmyexternalproxy
     ports:
-      - "1236:1236"
+    # actual and only port facing any connection from outside 
+      - "1239:1236"
     volumes:
-      - ./nginx.conf:/etc/nginx/nginx.conf
-    restart: always
+      - ./nginx.conf:/etc/nginx/nginx.conf:ro
+      - ./nginx_proxy.conf:/etc/nginx/proxy.conf:ro
+    restart: unless-stopped
     depends_on:
-      - pictrs
+      - lemmy-pictrs
       - lemmy-ui
 
   lemmy:
-    image: lemmy-dev:latest
-    ports:
-      - "8536:8536"
-      - "6669:6669"
-    restart: always
+    image: dessalines/lemmy:dev
+    container_name: lemmy
+    hostname: lemmy
```

Comment:
Why set the container name and hostname explicitly? Seems unnecessarily complex, plus it will probably prevent you from running multiple instances of this setup in different folders.

Reply:
Because your nginx will break if the hostname is not lemmy anymore. The routing won't work. We can remove container_name though, you're right!